src/classes/CRLP_RollupProcessor.cls

```diff
@@ -546,5 +590,21 @@ public inherited sharing class CRLP_RollupProcessor {
         return handler;
     }
 
+    /**
+     * @description Initializes rollupsDefsByParentId with external rollup data
+     */
+    private void initializeExternalData() {
+        if (!shouldFetchExternalData) {
+            return;
+        }
+
+        Set<Id> recordIds = new Set<Id>();
+        for (SObject parentRecord : parentRecords) {
+            recordIds.add(parentRecord.Id);
+        }
+
+        rollupsDefsByParentId = apiService.getBaseRollupStateForRecords(recordIds, rollupType);
```

Comment:
If the `withRollupDefinitions()` method was used to inject rollup definitions into this process (such as in Skew Mode), it might only do so for some of the ParentId's being rolled up. For example, Skew Mode might pass in 10 Account Summary Records, where the first has Rollup Definitions from a previous process (child batch iteration). In this example, `withRollupDefinitions` would get just the definitions for the one parent record. 

There would still be a need to retrieve the external rollup states for the other Accounts that did not have rollup defintion data injected. I think this affects this method as well as the two `withRollupDefinitions` methods.

Reply:
ahh good point, I'll make that change.

@rponti-sforg If you're currently testing this, there might be an issue with skew mode that I need to fix. I'll tag you again when it's back up

Reply:
> ahh good point, I'll make that change.
> 
> @rponti-sforg If you're currently testing this, there might be an issue with skew mode that I need to fix. I'll tag you again when it's back up

@seanpat09 Do I or should I create a new scratch org to test with?